Core/tooling/build.sh

```diff
@@ -2,15 +2,50 @@
 
 set -euxo pipefail
 
+# fix openpgp in node_modules
+echo "Patching openpgp.js v5..."
+for f in openpgp.min.js openpgp.min.js.map openpgp.min.mjs openpgp.min.mjs.map openpgp.mjs;
+do
+  if [ -f node_modules/openpgp/dist/$f ]; then rm -f node_modules/openpgp/dist/$f ; fi
+  if [ -f node_modules/openpgp/dist/node/$f ]; then rm -f node_modules/openpgp/dist/node/$f ; fi
+done
+extra_exports="
+// -----BEGIN ADDED BY FLOWCRYPT----
+exports.readToEnd = readToEnd;
+exports.util = util;
```

Comment:
Similarly here, except we'll have to re-implement. Or copy with a link to the source.

It's the following methods, if I'm seeing correctly: 
 - in native-crypto.js `Uint8Array_to_str` and `b64_to_Uint8Array`
 - in actual code `uint8ArrayToString`

I see only one use which is here:

```
  public static random = () => { // eg TDW6-DU5M-TANI-LJXY
    return base64encode(util.uint8ArrayToString(secureRandomBytes(128)))
      .toUpperCase().replace(/[^A-Z0-9]|0|O|1/g, '').replace(/(.{4})/g, '$1-').substr(0, 19);
  }
```

And that is not even used, so let's remove it. That way, can also remove this `build.sh` update. (let me know if there are other usages I overlooked)

We may still later need that implementation for `native-crypto.js` but for this PR here, we can do without.

Reply:
Removed `random()`. So after this we really only have one extra export, which sounds not a big hack.

Reply:
This really has to go - fixing this is as easy as adding the separate dependency and defining the type for one method. There is no excuse for us to be adding shell scripts and patching things in node_modules if the same can be resolved through standard methods.

If there was no other way, that would be a different story.

Reply:
@tomholub Well, I have removed this patch, I've added `@openpgp/web-stream-tools` into package.json, I've tried to import `readToEnd()` from it. And now for this code:
```typescript
import { readToEnd } from '@openpgp/web-stream-tools';
```
I'm getting following error:
```
Could not find a declaration file for module '@openpgp/web-stream-tools'. '/home/ivan/pro/flowcrypt/flowcrypt-ios/Core/node_modules/@openpgp/web-stream-tools/lib/streams.js' implicitly has an 'any' type.
```
What I've tried:
https://javascript.plainenglish.io/how-to-write-typescript-types-for-a-javascript-library-e598b9eb8be7
But in `ambients.d.ts` for the code
```typescript
declare module "@openpgp/web-stream-tools" {
```
I'm getting error:
```
Invalid module name in augmentation. Module '@openpgp/web-stream-tools' resolves to an untyped module at '/home/ivan/pro/flowcrypt/flowcrypt-ios/Core/node_modules/@openpgp/web-stream-tools/lib/streams.js', which cannot be augmented.ts(2665)
```
Any ideas how to solve this?

Reply:
Basically I think my questions are:
- where to put file with type definitions? (exact directory)
- what should be file name?
- how to solve above error if it will happen if I create properly named file in the proper directory.

Reply:
Removed `ambients.d.ts`, placed npm-installed-like type defintion in the `./source/core/types/`. No effect. So I really don't know how to solve this.

Reply:
This is why I tell patching openpgp.js the way I did was significantly easier.

Reply:
One more update - tried to copy that into `node_modules/@types`. So I have `node_modules/@types/@openpgp/web-stream-tools`. But - this also didn't work. No more ideas.

Reply:
Updated build script with web-stream-tools patch.

Reply:
But as result ... unfortunately, it just doesn't work:
```
> flowcrypt-mobile-core@0.0.1 test
> ./node_modules/.bin/ava build/ts/test.js --verbose


  Uncaught exception in build/ts/test.js

  /home/ivan/pro/flowcrypt/flowcrypt-ios/Core/node_modules/@openpgp/web-stream-tools/lib/streams.js:1
  import { isNode, isStream, isArrayStream, isUint8Array, concatUint8Array } from './util';
  ^^^^^^

  SyntaxError: Cannot use import statement outside a module

  › SyntaxError: Cannot use import statement outside a module
  › Object.<anonymous> (build/ts/core/pgp-key.js:14:28)

  ✖ build/ts/test.js exited with a non-zero exit code: 1
```

My overall impression - web-stream-tools are in the too early stage. I strongly suggest to give up on them for now and just patch openpgp.js - which was already proven to work. Later on, when they get more mature - we can try them again.

Reply:
@tomholub - what do you think of above?

Reply:
Posted them PR with initial type definitions https://github.com/openpgpjs/web-stream-tools/pull/11